Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+        auto callback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            if (visitArgs.m_registry.Get(level, visitArgs.m_jsonKeyPath))
+            {
+                if (levelName.compare(level.c_str()) == 0)
+                {
+                    levelIndex = currentIndex;
+                    return AZ::SettingsRegistryInterface::VisitResponse::Done;
+                }
+            }
+
+            currentIndex++;
+            return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+        };
+
+        auto key = FixedValueString::format("%s/Levels", m_path.c_str());
+        AZ::SettingsRegistryVisitorUtils::VisitArray(*m_settingsRegistry, callback, key);
+        return levelIndex;
+    }
+
+    QualityCVarGroup::ValueType QualityCVarGroup::GetHighestQualityForSetting(AZStd::string_view path)
+    {
+        ValueType maxQualityLevel = -1;
+        auto callback = [&]([[maybe_unused]] const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            maxQualityLevel++;
+            return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+        };
+        AZ::SettingsRegistryVisitorUtils::VisitArray(*m_settingsRegistry, callback, path);
+        return maxQualityLevel;
+    }
+
+    void QualityCVarGroup::LoadQualityLevel(ValueType qualityLevel)
+    {
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+        using Type = AZ::SettingsRegistryInterface::Type;
+
+        if (qualityLevel >= m_numQualityLevels || qualityLevel < 0)
+        {
+            AZ_Warning("QualityCVarGroup",
+                       false,
+                       "Invalid quality level %lld requested for %s which only supports quality levels 0..%lld",
+                       qualityLevel,
+                       m_functor->GetName(),
+                       m_numQualityLevels - 1);
+            return;
+        }
+
+        auto key = FixedValueString::format("%s/Settings", m_path.c_str());
+        auto settingsVisitorCallback = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            if (visitArgs.m_type == Type::Object ||
+                visitArgs.m_type == Type::Null ||
+                visitArgs.m_type == Type::NoType)
+            {
+                AZ_Warning("QualityCVarGroup", false, "Invalid setting value type for %.*s, objects and null are not supported, only arrays, bool, string, or numbers.", AZ_STRING_ARG(visitArgs.m_fieldName));
+                return VisitResponse::Skip;
+            }
+
+            AZStd::string_view command = visitArgs.m_fieldName;
+            AZStd::string_view valueKey = visitArgs.m_jsonKeyPath;
+            if (visitArgs.m_type == Type::Array)
+            {
+                valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(valueKey), qualityLevel);

```

Comment:
This is a dangling reference.
The FixedValueString goes out of scope after line 139, so the valueKey is pointing to stack memory of indeterminate value.
```suggestion
            FixedValueString valueKey = visitArgs.m_jsonKeyPath;
            if (visitArgs.m_type == Type::Array)
            {
                valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(valueKey), qualityLevel);
```

Reply:
Good catch!